test/e2e/testing-manifests/storage-csi/mock/csi-mock-driver.yaml

```
@@ -20,7 +20,9 @@ spec:
             - "--csi-address=$(ADDRESS)"
             # Topology support is needed for the pod rescheduling test
             # ("storage capacity" in csi_mock_volume.go).
-            - "--feature-gates=Topology=true"
+            # HonorPVReclaimPolicy support is needed in 
+            # test/e2e/storage/csimock/csi_honor_pv_reclaim_policy.go
+            - "--feature-gates=Topology=true,HonorPVReclaimPolicy=true"
```

Would you be able to use https://github.com/carlory/kubernetes/blob/master/test/e2e/storage/utils/deployment.go#L170 to ensure enabling the feature gate on external-provisioner only if the feature gate is enabled on k/k? something like below when the mock csi driver is being started

```suggestion
           if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.HonorPVReclaimPolicy) {
           o.Features["csi-provisioner"] = []string{"HonorPVReclaimPolicy=true"}
}
```

---

example: https://github.com/carlory/kubernetes/blob/master/test/e2e/storage/drivers/csi.go#L673-L675